clients/drcachesim/tests/signal_invariants.c

```diff
@@ -38,6 +38,9 @@
 
 #ifndef ASM_CODE_ONLY /* C code */
 
+/* Enable asserts in release build testing too. */
+#    undef NDEBUG
```

Comment:
Can't we add remove NDEBUG from the add_exe impl too: https://github.com/DynamoRIO/dynamorio/blob/371926b82e9b0e1c9415333dc3cbeb06bfe0bfa4/suite/tests/CMakeLists.txt#L633

This would handle drmemtrace.signal_invariants without needing to modify source.

Reply:
This would require modifying the suite/tests/CMakeLists code and is beyond the scope of this PR.  I added a XXX comment in signal_invariants.c about it.